packages/store/src/types.js

```diff
@@ -184,7 +184,7 @@
  */
 
 /**
- * @template K
+ * @template {Key & object} K
```

Comment:
```suggestion
 * @template {Key} K
```
Not weak. Why `& object`? Seems wrong.